crates/store/re_log_encoding/src/decoder/streaming.rs

```diff
@@ -0,0 +1,371 @@
+                                Compression::Off => {
+                                    if unprocessed_bytes.len() < uncompressed_len + header_size {
+                                        self.expect_more_data = true;
+                                        Pin::new(&mut self.reader).consume(buf_length);
+
+                                        continue;
+                                    }
+
+                                    (
+                                        &unprocessed_bytes
+                                            [header_size..uncompressed_len + header_size],
+                                        uncompressed_len,
+                                    )
+                                }
+
+                                Compression::LZ4 => {
+                                    if unprocessed_bytes.len() < compressed_len + header_size {
+                                        // Not enough data to read the message, need to wait for more
+                                        self.expect_more_data = true;
+                                        Pin::new(&mut self.reader).consume(buf_length);
+
+                                        continue;
+                                    }
+
+                                    uncompressed
+                                        .resize(uncompressed.len().max(uncompressed_len), 0);
+                                    let data = &unprocessed_bytes
+                                        [header_size..compressed_len + header_size];
+                                    if let Err(err) =
+                                        lz4_flex::block::decompress_into(data, uncompressed)
+                                    {
+                                        return std::task::Poll::Ready(Some(Err(
+                                            DecodeError::Lz4(err),
+                                        )));
+                                    }
+
+                                    (&uncompressed[..], compressed_len)
+                                }
+                            };
+
+                            // decode the message
+                            let msg = rmp_serde::from_slice::<LogMsg>(data);
+
+                            match msg {
+                                Ok(msg) => (Some(msg), length + header_size),
+                                Err(err) => {
+                                    return std::task::Poll::Ready(Some(Err(
+                                        DecodeError::MsgPack(err),
+                                    )));
+                                }
+                            }
+                        }
+                        super::MessageHeader::EndOfStream => return std::task::Poll::Ready(None),
+                    }
+                }
+                crate::Serializer::Protobuf => {
+                    let header_size = std::mem::size_of::<file::MessageHeader>();
```

Comment:
I haven't followed the recent developments on the protobuf file format, and this is not new to this PR, but seeing a `mem_size_of` on a struct type that uses the Rust ABI makes me both very uneasy and 99% there's a nasty bug lurking somewhere.

Reply:
In this case it's 2x u64 fields, which is not problematic unless changed. Until this point, the encoding/decoding was fully local to the same file where it's defined, and the compiler would yell at you if you add/remove fields or change their types, pointing out all the places that need to be changed.